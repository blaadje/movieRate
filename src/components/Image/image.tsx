import * as React from 'react'

interface iProps extends React.HTMLAttributes<any> {
  wrapperclass?: string,
  className?: string,
  filter?: boolean,
  filterClass?: string,
  src: string,
  children?: React.ReactNode,
}

const Image: React.SFC<iProps> = (props: iProps) => {
  const { src, children, className, filter, filterClass, ...htmlProps } = props
  return (
    <div {...htmlProps} className={`Image-wrapper ${className || ''}`} style={{ background: `url(${props.src})` }}>
      <div className={`${filter ? 'isShowed' : ''} ${filterClass || ''} Image-filter`}>
        {props.children}
      </div>
    </div>
  )
}

export default Image
