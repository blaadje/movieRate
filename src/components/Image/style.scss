.Image-wrapper {
  position: relative;
  background-repeat: no-repeat !important;
  background-size: cover !important;
}

.Image-filter {
  width: 100%;
  height: 100%;
  background: rgba(#176087, 0.2);
}