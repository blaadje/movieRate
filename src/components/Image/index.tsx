import * as React from 'react'

import './style.scss'

interface iProps extends React.HTMLAttributes<any> {
  className?: string,
  filter?: boolean,
  filterClass?: string,
  src: string,
  children?: React.ReactNode,
}

const Image: React.SFC<iProps> = (props: iProps) => {
  const { src, children, className, filter, filterClass, ...htmlProps } = props

  return (
    <div {...htmlProps} className={`Image-wrapper ${className}`} style={{ background: `url(${props.src})` }}>
      <div className={`${filter ? 'isShowed' : ''} ${filterClass || ''} Image-filter`}>
        {props.children}
      </div>
    </div>
  )
}

Image.defaultProps = {
  filter: false
}

export default Image
