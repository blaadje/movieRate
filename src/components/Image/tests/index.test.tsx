import Image from 'components/Image'

import * as React from 'react'
const renderer = require('react-test-renderer')

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Image
        className='bar'
        src='foo'
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
