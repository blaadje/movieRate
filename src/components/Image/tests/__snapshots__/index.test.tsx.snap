// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Image-wrapper bar"
  style={
    Object {
      "background": "url(foo)",
    }
  }
>
  <div
    className=" undefined"
  />
</div>
`;
